frontend/components/AllQuestions.jsx

```diff
@@ -1,69 +1,19 @@
-import { Box, Heading, Text, Stack, Grid } from "@chakra-ui/react";
-import { useState, useEffect } from "react";
-import axios from "axios";
+import api from "@/services/api";
+import { Box, Grid, Heading, Text } from "@chakra-ui/react";
+import { useQuery } from "react-query";
+import QuestionCard from "./QuestionCard";
 
-const AllQuestions = () => {
-  const [questions, setQuestions] = useState([]);
-
-  useEffect(() => {
-    async function fetchData() {
-      const result = await axios.get("http://localhost:8080/api/questions");
-      console.log(result.data.data)
-      const questionsWithMentors = await Promise.all(result.data.data.map(async (question) => {
-        const mentorId = question.mentorId;
-        if (!mentorId) return question;
-        const mentor = await axios.get(`http://localhost:8080/api/mentor/${mentorId}`);
-        return { ...question, mentorName: mentor.data.data.firstName + " " + mentor.data.data.lastName};
-      }));
-      setQuestions(questionsWithMentors);
-    }
-
-    fetchData();
-  }, []);
+const AllQuestions = ({ initialQuestions }) => {
+  const { data: questions } = useQuery({
```

Comment:
[React Query](https://tanstack.com/query/latest) is a commonly used data fetching library. It supports "caching, background updates and stale data out of the box with zero-configuration"